t/plugin/recaptcha.t

```diff
@@ -0,0 +1,241 @@
+                                  "response": {
+                                    "content_type": "application/json; charset=utf-8",
+                                    "status_code": 400,
+                                    "body": "{\"message\":\"invalid captcha\"}\n"
+                                  },
+                                  "ssl_verify": false
+                              }
+                           },
+                           "upstream": {
+                               "nodes": {
+                                   "127.0.0.1:1980": 1
+                               },
+                               "type": "roundrobin"
+                           },
+                           "uri": "/login"
+                   }]=]
+                   )
+               if code >= 300 then
+                   ngx.status = code
+                   ngx.say(body)
+               end
+
+               code, body = t('/apisix/admin/routes/2',
+                   ngx.HTTP_PUT,
+                   [=[{
+                          "plugins": {
+                             "recaptcha": {
+                                 "secret_key": "6LeIxAcTAAAAAGG-vFI1TnRWxMZNFuojJ4WifJWe",
+                                 "parameter_source": "query",
+                                 "parameter_name": "captcha",
+                                 "response": {
+                                   "content_type": "application/json; charset=utf-8",
+                                   "status_code": 400,
+                                   "body": "{\"message\":\"invalid captcha\"}\n"
+                                 },
+                                 "ssl_verify": false
+                             }
+                          },
+                          "upstream": {
+                              "nodes": {
+                                  "127.0.0.1:1980": 1
+                              },
+                              "type": "roundrobin"
+                          },
+                          "uri": "/active"
+                  }]=]
+                  )
+               if code >= 300 then
+                   ngx.status = code
+               end
+
+               ngx.say(body)
+           }
+       }
+--- response_body
+passed
+
+
+
+=== TEST 4: request is terminated by recaptcha plugin 1
+--- request
+POST /login
+--- error_code: 400
+--- response_headers
+Content-Type: application/json; charset=utf-8
+--- response_body
+{"message":"invalid captcha"}
+
+
+
+=== TEST 5: request is terminated by recaptcha plugin 2
+--- request
+POST /active
+--- error_code: 400
+--- response_headers
+Content-Type: application/json; charset=utf-8
+--- response_body
+{"message":"invalid captcha"}
+
+
+
+=== TEST 6: recaptcha valid
+--- request
+POST /login
+--- more_headers
+captcha: test
+--- error_code: 404
+
+
+
+=== TEST 7: recaptcha valid
+--- request
+POST /active?captcha=test
+--- error_code: 404
+--- no_error_log
```

Comment:
We don't need this?

Reply:
If you are talking about 
```
--- no_error_log
[error]
```
then yes.

If you are talking `TEST 6` and `TEST 7`. We need these cases for test passed cases. (because the test data had been added on `TEST4`, so I just omit it.)

Reply:
Yes, I just mean the `--- no_error_log`.